.github/workflows/ci-cd-workflow.yml

```diff
@@ -0,0 +1,248 @@
+name: pymagicc CI-CD
+on: push
+
+jobs:
+  linting-and-docs:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dev dependencies
```

Comment:
Couldn't this also use the cache?

Reply:
yep do you want to implement?

Reply:
:-) Will do after lunch!

Reply:
Ok, this didn't speed things up much it seems, but still hopefully the right thing (and reducing PyPA network bill).